recommendation_v2/torchrec_dlrm/dlrm_main.py

```diff
@@ -788,10 +788,17 @@ def main(argv: List[str]) -> None:
     # the optimizer update will be applied in the backward pass, in this case through a fused op.
     # TorchRec will use the FBGEMM implementation of EXACT_ADAGRAD. For GPU devices, a fused CUDA kernel is invoked. For CPU, FBGEMM_GPU invokes CPU kernels
     # https://github.com/pytorch/FBGEMM/blob/2cb8b0dff3e67f9a009c4299defbd6b99cc12b8f/fbgemm_gpu/fbgemm_gpu/split_table_batched_embeddings_ops.py#L676-L678
+
+    # Note that lr_decay and initial_accumulator_value for Adagrad cannot be specified below (FBGEMM version 0.3.2).
+    # This equivalently means that both parameters are hardcoded to zero.
     apply_optimizer_in_backward(
         embedding_optimizer,
         train_model.model.sparse_arch.parameters(),
-        {"lr": args.learning_rate},
+        {
+            "lr": args.learning_rate,
+            "weight_decay": WEIGHT_DECAY,
```

Comment:
I think you also need ot set 'weight_decay_mode' cc @jianyu

Reply:
To make weight_decay value effective, we can set `weight_decay_mode` to be `WeightDecayMode.L2". CC @csmiler

Reply:
I tried setting that but I this case I get the following error from [this line](https://github.com/pytorch/torchrec/blob/v0.3.2/torchrec/optim/apply_optimizer_in_backward.py#L51):
```
dlrm_main/0 [0]:  File "/code/dlrm_main.py", line 743, in <module>
dlrm_main/0 [0]:    main(sys.argv[1:])
dlrm_main/0 [0]:  File "/code/dlrm_main.py", line 665, in main
dlrm_main/0 [0]:    apply_optimizer_in_backward(
dlrm_main/0 [0]:  File "/opt/conda/lib/python3.10/site-packages/torchrec/optim/apply_optimizer_in_backward.py", line 80, in apply_optimizer_in_backward
dlrm_main/0 [0]:    _apply_optimizer_in_backward_to_param(param)
dlrm_main/0 [0]:  File "/opt/conda/lib/python3.10/site-packages/torchrec/optim/apply_optimizer_in_backward.py", line 51, in _apply_optimizer_in_backward_to_param
dlrm_main/0 [0]:    optimizer = optimizer_class([param], **optimizer_kwargs)
dlrm_main/0 [0]:TypeError: Adagrad.__init__() got an unexpected keyword argument 'weight_decay_mode'
```

* I'm using TorchRec == FBGEMM == 0.3.2 (see [here](https://github.com/mlcommons/training/blob/master/recommendation_v2/torchrec_dlrm/requirements.txt))
* If this doesn't work, that's not a problem for MLPerf: we're going to freeze `weight_decay==0` anyway. Of course it will be an issue for anyone using the code here and trying to change this parameter for whatever reason.

Reply:
it looks like for EXACT_ADAGRAD, weight_decay will not work w/o weight_decay_mode. This means that regardless of what this is set to, it will be a Noop

@janekl  I'd suggest  removing this line since it's a no-op anyways and we don't plan to use. We can just add to epsilon.

I'm going to add comments to fbgemm_gpu to specify the difference in eps default (which defaults to adam default) and that weight_decay is unsupported now for EXACT_ADAGRAD

Reply:
Thanks Colin, I've updated the PR according to your suggestion.